lib/src/search/widget/search_widget.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+
+import '../controller.dart';
+import 'search_navigation_widget.dart';
+import 'search_settings_widget.dart';
+
+const _iconSize = 24.0;
+
+class SearchWidget extends StatelessWidget {
+  final SearchController searchController;
+
+  const SearchWidget({
+    super.key,
+    required this.searchController,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: searchController,
+      builder: (context, child) => InkWell(
```

Comment:
Try a more intuitive solution with `GestureDetector` or something like that. If no better solution, then extract to a widget and comment on that widget why this is needed.